File: platform/app/schemas/agent.py
from __future__ import annotations

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class TokenUsage(BaseModel):
    input_tokens: int = 0
    output_tokens: int = 0
    total_tokens: int = 0
    estimated_cost_rmb: float = 0.0


class AgentStatusResponse(BaseModel):
    id: str
    role: str
    display_name: str
    status: str
    model_name: Optional[str] = None
    config: Optional[dict] = None
    current_task_id: Optional[str] = None
    started_at: Optional[datetime] = None
    last_active_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AgentListResponse(BaseModel):
    agents: List[AgentStatusResponse]


class AgentConfigUpdate(BaseModel):
    model_name: Optional[str] = None
    config: Optional[dict] = None
    # Accept frontend 'model' field as alias for model_name
    model: Optional[str] = None
    temperature: Optional[float] = None
    max_tokens: Optional[int] = None
    enabled: Optional[bool] = None

    def get_model_name(self) -> Optional[str]:
        """Return model_name, falling back to model alias."""
        return self.model_name or self.model


class AgentSessionResponse(BaseModel):
    role: str
    status: str
    current_task_id: Optional[str] = None
    uptime_seconds: Optional[float] = None
    token_usage: TokenUsage = TokenUsage()
    turns: int = 0

File: web/src/types/agent.ts
export interface TokenUsage {
  input_tokens: number;
  output_tokens: number;
  total_tokens: number;
  estimated_cost_rmb: number;
}

export interface AgentStatus {
  id: string;
  role: string;
  display_name: string;
  status: string;
  model_name: string | null;
  config: Record<string, unknown> | null;
  current_task_id: string | null;
  started_at: string | null;
  last_active_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface AgentListResponse {
  agents: AgentStatus[];
}

export interface AgentConfigUpdate {
  model_name?: string;
  config?: Record<string, unknown>;
  temperature?: number;
  max_tokens?: number;
  max_turns?: number;
  thinking_level?: string;
  extra_skill_dirs?: string[];
  system_prompt_append?: string;
  enabled?: boolean;
}


export interface AgentConfigOptions {
  available_models: string[];
  thinking_levels: string[];
  role_defaults: Record<string, string>;
}

export interface AgentConfigFormValues {
  model_name?: string;
  temperature?: number;
  max_tokens?: number;
  max_turns?: number;
  thinking_level?: string;
  extra_skill_dirs?: string[];
  system_prompt_append?: string;
  enabled?: boolean;
}
